File: CodingSample/DotNet/Fundamentals/Inheritance/ClassTypeTest/DemoApp1/Payroll/SalesPerson.cs
namespace Payroll
{
    //defining SalesPerson as a derived class of Employee (base class)
    class SalesPerson : Employee
    {
        //automatic property
        public double Sales { get; set; }

        public SalesPerson(int h, float r, double s) : base(h, r)
        {
            Sales = s;
        }

        //optional parameters of method may appear at the end of the
        //parameter list each initialized with a constant expression
        //which is implictly passed if the argument is not passed
        //explictly by the caller of the method
        public double Commission(double minimum, float percentage = 5)
        {
            return Sales < minimum ? 0 : Sales * percentage / 100;
        }

        public override double MonthlyIncome()
        {
            if(DailyWages == 0)
                return Commission(percentage: 15, minimum: 0);
            return base.MonthlyIncome() + Commission(20000);
        }
    }
}

File: CodingSample/Java/Core/Inheritance/SubclassTest/demo1/payroll/SalesPerson.java
package payroll;

//defining SalesPerson as a subclass of 
//Employee (super-class)
public class SalesPerson extends Employee {
    
    private double sales;

    public SalesPerson(int h, float r, double s) {
        super(h, r); //calling constructor of super-class
        sales = s;
    }

    public double getSales() {
        return sales;
    }

    public void setSales(double value) {
        sales = value;
    }

    //overriding method of super-class
    public double monthlyIncome() {
        double income = super.monthlyIncome();
        if(sales >= 20000)
            income += 0.05 * sales;
        return income;
    }
}
